generator/manipulation.go

```diff
@@ -0,0 +1,178 @@
+package generator
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"time"
+
+	"github.com/sirupsen/logrus"
+
+	"github.com/docker/oscalkit/types/oscal"
+	"github.com/docker/oscalkit/types/oscal/catalog"
+	"github.com/docker/oscalkit/types/oscal/profile"
+)
+
+//ProcessAddition processess additions of a profile
+func ProcessAddition(alt profile.Alter, controls []catalog.Control) []catalog.Control {
+	for j, ctrl := range controls {
+		if ctrl.Id == alt.ControlId {
+			for _, add := range alt.Additions {
+				for _, p := range add.Parts {
+					appended := false
+					for _, catalogPart := range ctrl.Parts {
+						if p.Class == catalogPart.Class {
+							appended = true
+							//append with all the parts with matching classes
+							parts := ModifyParts(p, ctrl.Parts)
+							ctrl.Parts = parts
+						}
+					}
+					if !appended {
+						ctrl.Parts = append(ctrl.Parts, p)
+					}
+				}
+			}
+			controls[j] = ctrl
+		}
+		for k, subctrl := range controls[j].Subcontrols {
+			if subctrl.Id == alt.SubcontrolId {
+				for _, add := range alt.Additions {
+					for _, p := range add.Parts {
+						appended := false
+						for _, catalogPart := range subctrl.Parts {
+							if p.Class == catalogPart.Class {
+								appended = true
+								//append with all the parts
+								parts := ModifyParts(p, subctrl.Parts)
+								subctrl.Parts = parts
+							}
+						}
+						if !appended {
+							subctrl.Parts = append(subctrl.Parts, p)
+						}
+					}
+
+				}
+			}
+			controls[j].Subcontrols[k] = subctrl
+		}
+	}
+	return controls
+}
+
+//ProcessAlteration processess alteration section of a profile
+func ProcessAlteration(alterations []profile.Alter, c *catalog.Catalog) *catalog.Catalog {
+	for _, alt := range alterations {
+		for i, g := range c.Groups {
+			c.Groups[i].Controls = ProcessAddition(alt, g.Controls)
+		}
+	}
+	return c
+}
+
+//ModifyParts modifies parts
+func ModifyParts(p catalog.Part, controlParts []catalog.Part) []catalog.Part {
+
+	//append with all the parts
+	var parts []catalog.Part
+	for i, part := range controlParts {
+		if p.Class != part.Class {
+			parts = append(parts, part)
+			continue
+		}
+		id := part.Id
+		part.Id = fmt.Sprintf("%s_%d", id, i+1)
+		parts = append(parts, part)
+		part.Id = fmt.Sprintf("%s_%d", id, i+2)
```

Comment:
@minhaj10p not sure I understand the logic behind this second part ID change

Reply:
This logic is for the scenario where if the `part`'s in the `add` section have the same `class` attribute, the `id`'s needed to be appended by an underscore and an increment